### ISSUE TITLE
Add support to have null models in diff editor to express fully added / fully deleted

### ISSUE THREAD

**Issue by nomeyer** [microsoft/monaco-editor#630]
Hey, first let me thank you for your hard work on this project – much appreciated.

I think the behaviour of the diff editor is incorrect for newly created files. See details below.

monaco-editor version: `0.10.1`

Browser: Chromium via electron `1.7.9`

OS: macOS Sierra `10.12.6`

Steps or JS usage snippet reproducing the issue:
```javascript
const editor = monaco.editor.createDiffEditor(someContainer, someOptions);
const modifiedModel = monaco.editor.createModel('teeeeest', 'javascript');
const originalModel = monaco.editor.createModel(null, 'javascript');
editor.setModel({ original: originalModel, modified: modifiedModel });
```

The diff displayed will show an empty line being deleted and a new line `teeeeest` being inserted. It should only display the new line being inserted – there is no empty line being deleted.

**Comment by alexandrudima** [comment]
I generally agree, but this is caused by ... convention.

Not only in the Monaco Editor, but in most editors, empty files are considered to consist of one line, with precisely one valid cursor position (1;1). Otherwise, there would be no chance to edit them and insert something...

The side-by-side diff editor is implemented by using two editors. By default the left-hand-side editor is read only, but the diff editor can be instantiated with an option to make both sides editable. It would then be very difficult to edit the left-hand-side (if it would contain no line).

Sorry for the weird explanation, probably not what you were expecting... :)

**Comment by nomeyer** [comment]
Thanks @alexandrudima.

This convention makes sense for creating / editing files. However, the convention for diffs when a file is created is definitely for the "before" side of the diff to be no content (as opposed to one empty line).

I can understand if this is not something you want to support though, given that this is primarily an editor and secondarily a diff viewer. It's just a shame for this pretty standard diff'ing scenario not to be covered.

**Comment by nomeyer** [comment]
Thanks for the acknowledgement @alexandrudima – I'm guessing this means you don't plan to support the diff behaviour I'm after, but just to be clear could you confirm please?

**Comment by chenmy0917** [comment]
I also need the feature mentioned in this issue, is it available in version 0.21.2? @alexdima 
I have read the issue [#1652](https://github.com/microsoft/monaco-editor/issues/1652) and used ```originalEditor.setHiddenAreas([new monaco.Range(1,1,5,1)])```, but it didn’t work.

![image](https://user-images.githubusercontent.com/22138217/94552991-64834900-028a-11eb-9761-417c4af29ed9.png)

**Comment by hediet** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

---

You might have more luck requesting this feature for VS Code - if we see a real need for VS Code, monaco editor will have the feature too.
